Fix -Wimplicit-function-declaration error with gcc 14.

Error:

```
arc4random.c:19:25: error: implicit declaration of function 'time'
[-Wimplicit-function-declaration]
   19 |                 srandom(time(NULL));
      |                         ^~~~
```

--- scanssh-2.1.3.1-origin/arc4random.c
+++ scanssh-2.1.3.1/arc4random.c
@@ -1,5 +1,6 @@
 #include <sys/types.h>
 #include <stdlib.h>
+#include <time.h>
 
 #include "config.h"
 
